unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -95,6 +95,34 @@ public static IntPtr gchandle_get_target_v2([NativeWrapperType("uintptr_t")] Int
         return Unsafe.As<object, IntPtr>(ref obj);
     }
 
+    [return: NativeCallbackType("MonoObject*")]
+    public static IntPtr object_isinst([NativeCallbackType("MonoObject*")] IntPtr obj, [NativeCallbackType("MonoClass*")] IntPtr klass)
+    {
+        var instance = Unsafe.As<IntPtr, object>(ref obj);
+        var type = Unsafe.As<IntPtr, Type>(ref klass);
+        Type current = instance.GetType();
+
+        while (current != null)
```

Comment:
@joncham @UnityAlex I recall we discussed for some methods there will be a slow way and maybe faster ways.   Does this reflection based implementation fall under the "slow way category"?  Is there a different approach to take that is faster?

Reply:
You may be able to call `Type.IsAssignableFrom` https://learn.microsoft.com/en-us/dotnet/api/system.type.isassignablefrom?view=net-7.0

Reply:
I believe the performance concern was more focused on object creation. The theory was that doing object creation through reflection would be slower than trying to stick to native or some other approach.